WebApplicationDemo/Application/TestContextAsync.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Ark.Tools.Outbox.SqlServer;
+using Ark.Tools.Sql;
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WebApplicationDemo.Application
+{
+    public class TestContextAsync : SqlContextAsyncFactory<ITestContext>
+    {
+        public TestContextAsync(IDataContextConfig contextConfig, IDbConnectionManager dbConnectionManager) : 
+            base(dbConnectionManager, contextConfig.SqlConnectionString)
+        {
+        }
+
+        public override ITestContext Create(DbConnection dbConnection, DbTransaction dbTransaction)
+        {
+            var ctx = new TestContext(dbConnection, dbTransaction);
+
+            return ctx;
+        }
+    }
+
+
+    public interface ITestContext : ISqlContextAsync, IAsyncDisposable
+    { 
+
+    }
+
+    public class TestContext : ITestContext
```

Comment:
this should inherith from a base class, AbstractSqlContext<tag> as currently does